src/midi/midi.cpp

```diff
@@ -149,6 +149,21 @@ class MidiState {
 		note_on_tracker.fill(false);
 	}
 
+	void Track(const MidiMessage& msg)
+	{
+		const auto status  = get_midi_status(msg.status());
+		const auto channel = get_midi_channel(msg.status());
+
+		if (status == MidiStatus::NoteOn) {
+			const auto note = msg.data1();
+			SetNoteActive(channel, note, true);
+
+		} else if (status == MidiStatus::NoteOff) {
+			const auto note = msg.data1();
+			SetNoteActive(channel, note, false);
```

Comment:
(ignore if this is refactored further)

When I see something like:

```c++
if (status == MidiStatus::NoteOn)
   SetNoteActive(…, true);
else if (status == MidiStatus::NoteOff)
   SetNoteActive(…,false);
```

If `SetNoteActive` took in `MidiStatus`, that could collapse to:
```c++
   SetNoteStatus(…, status);
```

or if we're just toggling the note off/on based on status:
```c++
   MaybeToggleNote(…, status);
```

Reply:
1. I agree with your comment in general, but not in this case 😄 

This is more like a switch/case block; I just didn't want to use that because then I'm forced to include a default branch which doesn't make sense here. I want to make it clear here how each different MIDI message type is handled. Merging two branches just to save a line or two is not desirable, as I see it. I prefer the clarity of expressions that we get by having separate branches.

```cpp
if (status == MidiStatus::NoteOn) {
	...

} else if (status == MidiStatus::NoteOff) {
	...

} else if (status == MidiStatus::ControlChange) {
	...
}
```

2. `SetNoteActive` just sets the note status to on or off in the note tracker. I should not have any knowledge about MIDI status codes! Again, I agree with you that boolean function args are undesirable, however, in such cases when it's quite obvious what they represent (an on/off switch, basically), I think they're ok.

Reply:
>  sets the "note active" flag to on or off in the note tracker. 

Thanks for this explanation, @johnnovak - and now it makes perfect sense "in name, as well "note tracker"!. Thanks!